### PR TITLE
feat(transaction): implement transaction deleter and test

### PR DESCRIPTION
**Pull request by JulienGrob**

### CODE REVIEW

File: src/account/application/transaction/deleter.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from src.account.application.transaction.repository import TransactionRepository, TransactionNotFound
+from src.account.domain.account import AccountId
+from src.account.domain.transaction import TransactionId
+from src.shared.application.repository import EntityNotFound
+
+
+@dataclass(frozen=True)
+class TransactionDeletionRequest:
+    id: TransactionId
+
```
Comment (JulienGrob) [inline src/account/application/transaction/deleter.py:13]: account_id: AccountId

File: src/account/application/transaction/deleter.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+from src.account.application.transaction.repository import TransactionRepository, TransactionNotFound
+from src.account.domain.account import AccountId
+from src.account.domain.transaction import TransactionId
+from src.shared.application.repository import EntityNotFound
+
+
+@dataclass(frozen=True)
+class TransactionDeletionRequest:
```
Comment (JulienGrob) [inline src/account/application/transaction/deleter.py:11]: More than id of the transaction is it useful and essential to add an account_id to the TransactionDeletionRequest ?

Comment (gmonacho) [reply src/account/application/transaction/deleter.py:11]: Why could it be usefull ? :thought_balloon:

Comment (gmonacho) [reply src/account/application/transaction/deleter.py:11]: Entities' identifiers are unique. It means that when you manipulate a transaction ID, it belongs to only one transaction, The account's id is useless here. The user is supposed to know which account his transaction is linked to.

**Review by gmonacho — APPROVED** [review]

**Pull request merged**